File: test_debug_logs.py
# Test file for Python print() debug detection

import os
# import logging # Example of using proper logging

# logging.basicConfig(level=logging.INFO)

def load_config():
    api_key = os.environ.get("API_KEY", "default_key")
    print(f"API Key loaded: {api_key}") # This should be flagged
    # logging.info(f"API Key loaded (first 5 chars): {api_key[:5]}...") # Safer logging example
    return api_key

def process_data(data):
    print("Starting data processing...") # This should be flagged
    processed_value = data * 2
    if processed_value > 10:
        print(f"Processed value {processed_value} is large!") # Indented print, should be flagged
    
    # print("Processing finished.") # Commented out, should NOT be flagged
    return processed_value

config = load_config()
result = process_data(5)
print(f"Final result: {result}") # This should be flagged

File: test_debug_logs.js
// Test file for console.log debug detection

function processUserData(user) {
  console.log('Processing user:', user.id); // This should be flagged

  const importantData = user.secretToken; // Example sensitive data
  // This is bad - logging sensitive data!
  console.log(`User token found: ${importantData}`); // This should be flagged
  
  // Some logic here...
  const result = importantData.toUpperCase();

  // console.log('Processing complete for user:', user.id); // This is commented out, should NOT be flagged

  return result;
}

function setupConfig() {
    const configValue = process.env.CONFIG_VAR || 'default';
    console.log("Config value set to:", configValue); // This should be flagged
}

const sampleUser = { id: 'user123', secretToken: 'tok_abc123xyz' };
processUserData(sampleUser);
setupConfig();

// Use a proper logger in production!
// import logger from './logger';
// logger.info('User processed', { userId: user.id });
